web3/middleware/validation.py

```diff
@@ -33,7 +33,6 @@ def transaction_normalizer(transaction):
 
 def validation_middleware(make_request, web3):
     transaction_validator = apply_formatters_to_dict({
-        'chainId': validate_chain_id(web3),
```

Comment:
I see that this is the cause for using xfail! 
Read the description in the PR.  However If `validate_chain_id` is not needed anymore, why is the code left behind?
Is this something you plan to return back to

Reply:
> Is this something you plan to return back to?

Yes. Once we can reliably discover the chainID from the connected node, then we should validate chainID again.